### ISSUE TITLE
Train the base models with new sliding windows.

### ISSUE THREAD

**Issue by AndreaBasile97** [AndreaBasile97/EnergyConsumption#11]
Cambiare le fold in modo tale che siano le stesse usate dal miglior setting MTGNN E rieffettuare il training.

Setup:
(.venv) PS E:\EnergyProject-main> python main.py --pathCSV "energy_consumption.csv" --config MULTI-STEP --target grid --histFeatures "grid,solar" --key house --dateCol date --windowSize 12  --numTargets 4 --spatial "PCNM"
energy_consumption.csv MULTI-STEP grid ['grid', 'solar'] house date 12 4
[Int64, Datetime(time_unit='ms', time_zone='UTC'), Float64, Float64, Utf8, Int64, Int64, Float64, Float64, Int32, UInt32]
Applying PCNM
R[write to console]: Loading required package: permute

R[write to console]: Loading required package: lattice

R[write to console]: This is vegan 2.6-4

configuration: MULTI-STEP_PCNM
monthly predictions
[array([5, 6, 7], dtype=uint32), array([6, 7, 8], dtype=uint32), array([7, 8, 9], dtype=uint32), array([ 8,  9, 10], dtype=uint32), array([ 9, 10, 11], dtype=uint32)]
first date of traning set: 2019-05-01 08:00:00+00:00
last date of traning set: 2019-06-30 20:00:00+00:00
first date of test set: 2019-07-01 00:00:00+00:00
last date of test set: 2019-07-31 23:45:00+00:00
first date of traning set: 2019-06-01 00:00:00+00:00
last date of traning set: 2019-07-31 20:00:00+00:00
first date of test set: 2019-08-01 00:00:00+00:00
last date of test set: 2019-08-31 23:45:00+00:00
first date of traning set: 2019-07-01 00:00:00+00:00
last date of training set: 2019-08-31 20:00:00+00:00
first date of testset: 2019-09-01 00:00:00+00:00
last date of testset: 2019-09-30 23:45:00+00:00
first date of traning set: 2019-08-01 00:00:00+00:00
last date of traning set: 2019-09-30 20:00:00+00:00
first date of test set: 2019-10-01 00:00:00+00:00
last date of test set: 2019-10-31 23:45:00+00:00
first date of traning set: 2019-09-01 00:00:00+00:00
last date of traning set: 2019-10-31 20:00:00+00:00
first date of test set: 2019-11-01 00:00:00+00:00
last date of test set: 2019-11-01 04:00:00+00:00
```

**Comment by AndreaBasile97** [comment]
# XGBooster 12-4
```
(.venv) PS E:\EnergyProject-main> python main.py --pathCSV "energy_consumption.csv" --config MULTI-STEP --target grid --histFeatures "grid,solar" --key house --dateCol date --windowSize 12  --numTargets 4 --spatial "PCNM"
energy_consumption.csv MULTI-STEP grid ['grid', 'solar'] house date 12 4
[Int64, Datetime(time_unit='ms', time_zone='UTC'), Float64, Float64, Utf8, Int64, Int64, Float64, Float64, Int32, UInt32]
Applying PCNM
R[write to console]: Loading required package: permute

R[write to console]: Loading required package: lattice

R[write to console]: This is vegan 2.6-4

configuration: MULTI-STEP_PCNM
monthly predictions
[array([5, 6, 7], dtype=uint32), array([6, 7, 8], dtype=uint32), array([7, 8, 9], dtype=uint32), array([ 8,  9, 10], dtype=uint32), array([ 9, 10, 11], dtype=uint32)]
first date of traning set: 2019-05-01 08:00:00+00:00
last date of traning set: 2019-06-30 20:00:00+00:00
first date of test set: 2019-07-01 00:00:00+00:00
last date of test set: 2019-07-31 23:45:00+00:00
first date of traning set: 2019-06-01 00:00:00+00:00
last date of traning set: 2019-07-31 20:00:00+00:00
first date of test set: 2019-08-01 00:00:00+00:00
last date of test set: 2019-08-31 23:45:00+00:00
first date of traning set: 2019-07-01 00:00:00+00:00
last date of training set: 2019-08-31 20:00:00+00:00
first date of testset: 2019-09-01 00:00:00+00:00
last date of testset: 2019-09-30 23:45:00+00:00
first date of traning set: 2019-08-01 00:00:00+00:00
last date of traning set: 2019-09-30 20:00:00+00:00
first date of test set: 2019-10-01 00:00:00+00:00
last date of test set: 2019-10-31 23:45:00+00:00
first date of traning set: 2019-09-01 00:00:00+00:00
last date of traning set: 2019-10-31 20:00:00+00:00
first date of test set: 2019-11-01 00:00:00+00:00
last date of test set: 2019-11-01 04:00:00+00:00
```

Results:

```
┌──────┬───────┬───────┬───────────────────────────────────┬─────────────────┬───────────────────────────────────┐
│ rmse ┆ rse   ┆ r2    ┆ window                            ┆ conf            ┆ method                            │
│ ---  ┆ ---   ┆ ---   ┆ ---                               ┆ ---             ┆ ---                               │
│ f64  ┆ f64   ┆ f64   ┆ str                               ┆ str             ┆ str                               │
╞══════╪═══════╪═══════╪═══════════════════════════════════╪═════════════════╪═══════════════════════════════════╡
│ 0.78 ┆ 0.252 ┆ 0.748 ┆ [array([5, 6, 7], dtype=uint32),… ┆ MULTI-STEP_PCNM ┆ XGBRegressor(base_score=None, bo… │
└──────┴───────┴───────┴───────────────────────────────────┴─────────────────┴───────────────────────────────────┘
```

**Comment by AndreaBasile97** [comment]
# MLPRegressor

Setting:
```
                MLPRegressor(hidden_layer_sizes=(128, 128, 128), activation='relu', solver='adam', 
                                alpha=0.0001, batch_size='auto', learning_rate='constant', 
                                learning_rate_init=0.001, power_t=0.5, max_iter=1000, shuffle=True, 
                                random_state=None, tol=0.0001, verbose=False, warm_start=False, 
                                momentum=0.9, nesterovs_momentum=True, early_stopping=False, 
                                validation_fraction=0.1, beta_1=0.9, beta_2=0.999, epsilon=1e-08, 
                                n_iter_no_change=10),
```

```
┌───────┬───────┬───────┬───────────────────────────────────┬─────────────────┬───────────────────────────────────┐
│ rmse  ┆ rse   ┆ r2    ┆ window                            ┆ conf            ┆ method                            │
│ ---   ┆ ---   ┆ ---   ┆ ---                               ┆ ---             ┆ ---                               │
│ f64   ┆ f64   ┆ f64   ┆ str                               ┆ str             ┆ str                               │
╞═══════╪═══════╪═══════╪═══════════════════════════════════╪═════════════════╪═══════════════════════════════════╡
│ 0.843 ┆ 0.306 ┆ 0.694 ┆ [array([5, 6, 7], dtype=uint32),… ┆ MULTI-STEP_PCNM ┆ MLPRegressor(hidden_layer_sizes=… │
└───────┴───────┴───────┴───────────────────────────────────┴─────────────────┴───────────────────────────────────┘
```

**Comment by AndreaBasile97** [comment]
# Random Forest

Setting:

```
RandomForestRegressor(n_jobs=-1, random_state=1)
```


```
┌───────┬───────┬───────┬───────────────────────────────────┬─────────────────┬───────────────────────────────────┐
│ rmse  ┆ rse   ┆ r2    ┆ window                            ┆ conf            ┆ method                            │
│ ---   ┆ ---   ┆ ---   ┆ ---                               ┆ ---             ┆ ---                               │
│ f64   ┆ f64   ┆ f64   ┆ str                               ┆ str             ┆ str                               │
╞═══════╪═══════╪═══════╪═══════════════════════════════════╪═════════════════╪═══════════════════════════════════╡
│ 0.792 ┆ 0.259 ┆ 0.741 ┆ [array([5, 6, 7], dtype=uint32),… ┆ MULTI-STEP_PCNM ┆ RandomForestRegressor(n_jobs=-1,… │
└───────┴───────┴───────┴───────────────────────────────────┴─────────────────┴───────────────────────────────────┘
```

**Comment by AndreaBasile97** [comment]
# KNeighborg

```
KNeighborsRegressor(n_neighbors=3, n_jobs=-1)
```


```
┌──────┬───────┬───────┬───────────────────────────────────┬─────────────────┬───────────────────────────────────┐
│ rmse ┆ rse   ┆ r2    ┆ window                            ┆ conf            ┆ method                            │
│ ---  ┆ ---   ┆ ---   ┆ ---                               ┆ ---             ┆ ---                               │
│ f64  ┆ f64   ┆ f64   ┆ str                               ┆ str             ┆ str                               │
╞══════╪═══════╪═══════╪═══════════════════════════════════╪═════════════════╪═══════════════════════════════════╡
│ 0.93 ┆ 0.347 ┆ 0.653 ┆ [array([5, 6, 7], dtype=uint32),… ┆ MULTI-STEP_PCNM ┆ KNeighborsRegressor(n_jobs=-1, n… │
└──────┴───────┴───────┴───────────────────────────────────┴─────────────────┴───────────────────────────────────┘
```
